tests/php/TestCase.php

```diff
@@ -0,0 +1,99 @@
+<?php # -*- coding: utf-8 -*-
+
+namespace Mollie\WooCommerce\Tests;
+
+use function Brain\Monkey\setUp;
+use function Brain\Monkey\tearDown;
+use Mockery;
+use PHPUnit\Framework\TestCase as PhpUniTestCase;
+use PHPUnit_Framework_MockObject_MockBuilder;
+use ReflectionException;
+use ReflectionMethod;
+
+/**
+ * Class Testcase
+ */
+class TestCase extends PhpUniTestCase
```

Comment:
May you elaborate more? Why a trait?

Reply:
Well, all the reasons of horizontal re-use, really. What are you going to do when you start using another library, and that library requires your test cases to extend _their_ class?

Reply:
Well, in that case I just need to change the class with the appropiated methods.

Also keep in mind we use BrainMonkey that need to do stuffs within the `setUp` and `tearDown` methods. So you'll still need to have to extend the main php unit class unless you want to duplicate the same logic on every test class. 

You say 

> What are you going to do when you start using another library,

What library a replacement for PhpUnit?

Reply:
There seems to be some confusion. Of course, you will need to extend PHPUnit's test case class. In fact, I don't think there's another way to make PHPUnit tests. What I'm saying is: extend PHPUnit's class, unless you absolutely **have** to extend something else, because when that something else comes (and I mean in addition, not as a replacement for PHPUnit), then you will have to choose what to extend, because you can't extend two things (your base class, and the new thing's base class) at the same time.

Reply:
Since you have to extend Phpunit class I don't see any problem in adding additional methods in it, them can be "if needed" moved into a "Trait" without any problem in future. 

In addition of PHPUnit means:

1. Php Unit do a BC change 
2. You want to add an extra layer of abstraction

You're classes depends on your custom _TestCase_ class and in case of addition you would have more problem to update all of the extended classes than having to move the methods which are within the test case class.

Reply:
I don't really see any more problems in updating the trait as opposed to updating the base class. Instead, it gives you more flexibility, i.e. you will be able to use a base class later with ease _if required_. For example, the xpmock lib includes a trait, but it could have required you to use their base class instead. In that case, you would have to chose which base class to extend. With traits, this problem would not present.

But again, if you strongly prefer to use a base class for now, then so be it.